internal/internal_workflow_testsuite.go

```diff
@@ -2541,6 +2550,24 @@ func (env *testWorkflowEnvironmentImpl) RegisterNexusAsyncOperationCompletion(
 	err error,
 	delay time.Duration,
 ) error {
+	opRef := env.nexusOperationRefs[service][operation]
```

Comment:
Better do this in two steps and provide better output in case the service isn't registered.

Also, why panic if this method returns an error? Is it standard in the rest of these `Register` methods? (didn't look into it, trust you to check).

Reply:
Only operations that are being mocked are being added to `nexusOperationRefs` map. So, if the user is not explicitly registering the service, and using `OnNexusOperation("service-name", ...)`, the service won't be registered (taking into consideration the other comment to not auto register services and operations).

Reply:
This response doesn't address my comment that we should either return an error and not panic (if consistent with other APIs on this struct) and if we do panic, we should provide a more helpful message.

Reply:
Oh, right. Here I can return an error. I'll update.